dev-packages/e2e-tests/test-applications/nestjs-with-submodules/tests/errors.test.ts

```diff
@@ -1,12 +1,125 @@
+  const errorEventPromise = waitForError('nestjs', event => {
+    return !event.type && event.exception?.values?.[0]?.value === 'This is an uncaught exception!';
+  });
+
+  const response = await fetch(`${baseURL}/example-module/unexpected-exception`);
+  expect(response.status).toBe(500);
+
+  const errorEvent = await errorEventPromise;
+
+  expect(errorEvent.exception?.values).toHaveLength(1);
+  expect(errorEvent.exception?.values?.[0]?.value).toBe('This is an uncaught exception!');
+
+  expect(errorEvent.request).toEqual({
+    method: 'GET',
+    cookies: {},
+    headers: expect.any(Object),
+    url: 'http://localhost:3030/example-module/unexpected-exception',
+  });
+
+  expect(errorEvent.transaction).toEqual('GET /example-module/unexpected-exception');
+
+  expect(errorEvent.contexts?.trace).toEqual({
+    trace_id: expect.any(String),
+    span_id: expect.any(String),
+  });
+});
+
+test('Sends unexpected exception to Sentry if thrown in module that was registered before Sentry', async ({
+  baseURL,
+}) => {
+  const errorEventPromise = waitForError('nestjs', event => {
+    return !event.type && event.exception?.values?.[0]?.value === 'This is an uncaught exception!';
+  });
+
+  const response = await fetch(`${baseURL}/example-module-wrong-order/unexpected-exception`);
+  expect(response.status).toBe(500);
+
+  const errorEvent = await errorEventPromise;
+
+  expect(errorEvent.exception?.values).toHaveLength(1);
+  expect(errorEvent.exception?.values?.[0]?.value).toBe('This is an uncaught exception!');
+
+  expect(errorEvent.request).toEqual({
+    method: 'GET',
+    cookies: {},
+    headers: expect.any(Object),
+    url: 'http://localhost:3030/example-module-wrong-order/unexpected-exception',
+  });
+
+  expect(errorEvent.transaction).toEqual('GET /example-module-wrong-order/unexpected-exception');
+
+  expect(errorEvent.contexts?.trace).toEqual({
+    trace_id: expect.any(String),
+    span_id: expect.any(String),
+  });
+});
+
+test('Custom global filter defined in module handles exception correctly', async ({ baseURL }) => {
+  let errorEventOccurred = false;
+
+  waitForError('nestjs', event => {
+    if (!event.type && event.exception?.values?.[0]?.value === 'Something went wrong in the example module!') {
+      errorEventOccurred = true;
+    }
+
+    return event?.transaction === 'GET /example-module/expected-exception';
+  });
+
+  const transactionEventPromise = waitForTransaction('nestjs', transactionEvent => {
+    return transactionEvent?.transaction === 'GET /example-module/expected-exception';
+  });
+
+  const response = await fetch(`${baseURL}/example-module/expected-exception`);
+  expect(response.status).toBe(400);
+
+  await transactionEventPromise;
+
+  await new Promise(resolve => setTimeout(resolve, 10000));
+
+  expect(errorEventOccurred).toBe(false);
+});
+
+test('Custom local filter defined in module handles exception correctly', async ({ baseURL }) => {
```

Comment:
m: Similar to above I would make this test description more specific.

Reply:
Updated name to "Does not send exception to Sentry if user-defined local exception filter already catches the exception"